engine/script/template/p2pkh_test.go

```diff
@@ -0,0 +1,117 @@
+package template
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/libsv/go-bk/bec"
+	"github.com/libsv/go-bk/crypto"
+	"github.com/libsv/go-bt/bscript"
+	assert "github.com/stretchr/testify/require"
+)
+
+func TestP2PKH(t *testing.T) {
+	validTests := []struct {
+		name     string
+		satoshis uint64
+		expected []P2PKHTemplate
+	}{
+		{
+			name:     "valid input",
+			satoshis: 1000,
+			expected: []P2PKHTemplate{
+				{
+					Script:   "76a9fd88ac",
+					Satoshis: 1000,
+				},
+			},
+		},
+		{
+			name:     "zero satoshis",
+			satoshis: 0,
+			expected: []P2PKHTemplate{
+				{
+					Script:   "76a9fd88ac",
+					Satoshis: 0,
```

Comment:
can we assume that actually 0 is also invalid value?
Because this doesn't make sense to have Pay To Public Key Hash of 0 satoshis.

Reply:
ok